src/app/model/Pawn.ts

```diff
@@ -0,0 +1,73 @@
+import { Board } from './Board';
+import { Piece } from './Piece';
+import { PieceMovement } from './PieceMovement';
+import { Column, columns, Row, Side, Square } from './Types';
+
+const NEXT_ROW_DIFFERENCE = {
+  WHITE: 1,
+  BLACK: -1,
+};
+
+export class Pawn extends Piece implements PieceMovement {
+  private nextRowDifference = 1;
+
+  constructor(id: string, side: Side) {
+    super('uselessId', side);
+    if (this.side !== 'WHITE') this.nextRowDifference = -1;
+    this.nextRowDifference = this.side === 'WHITE' ? NEXT_ROW_DIFFERENCE.WHITE : NEXT_ROW_DIFFERENCE.BLACK;
+  }
+
+  possibleMoves(position: Square, board: Board): Square[] {
+    let movesToGo: Square[] = [] as Square[];
+    movesToGo = movesToGo.concat(this.goAhead(position, board), this.goDoubleAhead(position, board), this.goDiagonalAhead(position, board));
+    return movesToGo;
+  }
+
+  private goAhead(position: Square, board: Board): Square[] {
+    const movesToGo: Square[] = [];
+    if (!board.onPositionPiece({ column: position.column, row: (position.row + 1 * this.nextRowDifference) as Row })) {
+      movesToGo.push({ column: position.column, row: (position.row + 1 * this.nextRowDifference) as Row });
+    }
+    return movesToGo;
+  }
+
+  private goDoubleAhead(position: Square, board: Board): Square[] {
+    const movesToGo: Square[] = [];
+    if ((position.row === 2 && this.side === 'WHITE') || (position.row === 7 && this.side === 'BLACK')) {
```

Comment:
Ważne jest opisowe nazywanie tego co robisz. Tzn. Tutaj sprawdzasz row 2 / side WHITE itp. ale dobrze to nazwac w terminach szachów, żeby ktoś od razu zrozumiał o co chodzi. Co powisz np. na wyciągnięcie to do metody np:
```
  private onStartingPosition(position: Square): boolean {
    return (position.row === 2 && this.side === 'WHITE') || (position.row === 7 && this.side === 'BLACK');
  }
```?